index.js

```diff
@@ -239,27 +243,35 @@ module.exports = class Filter extends Plugin {
         let operation = patch[0];
         let relativePath = patch[1];
         let entry = patch[2];
-        let outputPath = destDir + '/' + (this.getDestFilePath(relativePath, entry) || relativePath);
-        let outputFilePath = outputPath;
+        let fileMeta = this.fsMerger.readFileMeta(relativePath, { basePath: entry.basePath });
+        let outputPath = relativePath;
+        if (fileMeta && fileMeta.getDestinationPath) {
+          outputPath = fileMeta.getDestinationPath(relativePath);
+        } else {
+          if (fileMeta && fileMeta.prefix) {
+            outputPath = fileMeta.prefix + '/' + outputPath;
+          }
+        }
+        outputPath = this.getDestFilePath(outputPath, entry) || outputPath;
         let forceInvalidation = invalidated.includes(relativePath);
 
         this._logger.debug('[operation:%s] %s', operation, relativePath);
 
         switch (operation) {
           case 'mkdir': {
             instrumentation.mkdir++;
-            return fs.mkdirSync(outputPath);
+            return this.output.mkdirSync(outputPath, { recursive: true });
```

Comment:
This should not be recursive, the efficient patch based operation are crafted in such a way where no extra IO is required. Making this recursive can introduce unwanted IO, and also hide bugs where the list of operations we are applying was incomplete.

Reply:
@SparshithNR  ^^